src/python/pants/backend/docker/utils.py

```diff
@@ -12,16 +12,38 @@
 
 class KeyValueSequenceUtil(FrozenOrderedSet[str]):
     @classmethod
-    def from_strings(cls: type[_T], *strings: str) -> _T:
+    def from_strings(cls: type[_T], *strings: str, duplicates_must_match=False) -> _T:
         """Takes all `KEY`/`KEY=VALUE` strings and dedupes by `KEY`.
 
-        The last seen `KEY` wins in case of duplicates.
+        The last seen `KEY` wins in case of duplicates, unless `duplicates_must_match` is `True`, in
+        which case all `VALUE`s must be equal, if present.
         """
 
-        key_to_entry: dict[str, str] = {}
+        key_to_entry_and_value: dict[str, tuple[str, str | None]] = {}
         for entry in strings:
-            # Note that last entry with the same key wins.
-            key_to_entry[entry.partition("=")[0]] = entry
-        deduped_entries = sorted(key_to_entry.values())
-
+            key, has_value, value = entry.partition("=")
+            if not duplicates_must_match:
+                # Note that last entry with the same key wins.
+                key_to_entry_and_value[key] = (entry, value if has_value else None)
+            else:
+                prev_entry, prev_value = key_to_entry_and_value.get(key, (None, None))
+                if prev_entry is None:
+                    # Not seen before.
+                    key_to_entry_and_value[key] = (entry, value if has_value else None)
+                elif not has_value:
+                    # Seen before, no new value, so keep existing.
+                    pass
```

Comment:
How about using `continue`?

Reply:
potato tomato. End result is the same.. so 🤷🏽 

I prefer `continue` when I want to avoid the rest of the loop and start next iteration. So, in case the loop grows, It may or may not be appropriate with continue here... until then, I think it's a matter of taste, really (or perhaps it could potentially save one instruction cycle.. not sure)